From 1d5471ca0797a04744485fe72a7ea17ffd816e01 Mon Sep 17 00:00:00 2001
From: Devi Priya <quic_devipriy@quicinc.com>
Date: Sun, 17 Dec 2023 01:48:16 +0530
Subject: [PATCH] drivers: soc: qcom: Do ioremap instead of ioremap_resource

Add devm_ioremap instead of ioremap_resource to avoid ioremap failures
as llcc_base and llcc_broadcast_base are mapped to the same region.

Change-Id: Ib67ac8e744b0169a4e1babb0df06f0fff9f664c5
Signed-off-by: Devi Priya <quic_devipriy@quicinc.com>
---
 drivers/soc/qcom/llcc-qcom.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/qcom/llcc-qcom.c b/drivers/soc/qcom/llcc-qcom.c
index c61848595da0..8bf0234178da 100644
--- a/drivers/soc/qcom/llcc-qcom.c
+++ b/drivers/soc/qcom/llcc-qcom.c
@@ -917,6 +917,7 @@ static struct regmap *qcom_llcc_init_mmio(struct platform_device *pdev, u8 index
 					  const char *name)
 {
 	void __iomem *base;
+	struct resource *res;
 	struct regmap_config llcc_regmap_config = {
 		.reg_bits = 32,
 		.reg_stride = 4,
@@ -924,7 +925,13 @@ static struct regmap *qcom_llcc_init_mmio(struct platform_device *pdev, u8 index
 		.fast_io = true,
 	};
 
-	base = devm_platform_ioremap_resource(pdev, index);
+	res = platform_get_resource(pdev, IORESOURCE_MEM, index);
+
+	if (IS_ERR(res))
+		return ERR_CAST(res);
+
+	base = devm_ioremap(&pdev->dev, res->start,resource_size(res));
+
 	if (IS_ERR(base))
 		return ERR_CAST(base);
 
-- 
2.34.1

